### PR TITLE
Tool changes related to joinsplit inputs and outputs (depends on #379)

### PR DESCRIPTION
**Pull request by dtebbs**

### CODE REVIEW

File: client/zeth/core/mixer_client.py [hunk 1]
```diff
@@ -826,21 +810,31 @@ def _trap_r_randomness() -> str:
 
 def _compute_rho_i(phi: bytes, hsig: bytes, i: int) -> bytes:
     """
-    Returns rho_i = blake2s(0 || i || 10 || [phi]_252 || hsig)
+    Returns rho_i = blake2s(0 || (i..i) || 10 || [phi_truncated] || hsig)
```
Comment (AntoineRondelet) [inline client/zeth/core/mixer_client.py:813]: What is `i..i`? Isn't it `i`? (i.e. if `1..5` -> 1,2,3,4,5, then `i..i` -> i). What do I miss?

Comment (dtebbs) [reply client/zeth/core/mixer_client.py]: This is supposed to represent the bits in the binary representation of i.  I've adjusted the comments, so hopefully it's clearer now.  Thanks.

Comment (AntoineRondelet) [reply client/zeth/core/mixer_client.py]: I see, thanks that's clearer now

**Review by AntoineRondelet — APPROVED** [review]
LGTM, thanks @dtebbs

**Pull request merged**